emitter.py

```diff
@@ -124,13 +115,57 @@ def http_emitter(message, log, agentConfig, endpoint):
 
     except Exception:
         log.exception("Unable to post payload.")
-        try:
-            log.error("Received status code: {0}".format(r.status_code))
-        except Exception:
-            pass
 
 
-def post_headers(agentConfig, payload):
+def split_payload(legacy_payload):
+    metrics = list(legacy_payload['metrics'])
+    del legacy_payload['metrics']
+
+    metrics_payload = {"series": []}
+
+    # See https://github.com/DataDog/dd-agent/blob/5.11.1/checks/__init__.py#L905-L926 for format
+    for ts in metrics:
+        sample = {
+            "metric": ts[0],
+            "points": [[ts[1], ts[2]]]
+        }
+
+        if len(ts) >= 4:
+            if ts[3].get('type'):
+                sample['type'] = ts[3]['type']
+            if ts[3].get('hostname'):
+                sample['host'] = ts[3]['hostname']
+            if ts[3].get('tags'):
+                sample['tags'] = ts[3]['tags']
+            if ts[3].get('device_name'):
+                sample['device'] = ts[3]['device_name']
```

Comment:
Actually i want to keep this logic kinda isolated as it's a hack. Better to keep it in a single place to keep the code cleaner. Agent 6 will get rid of all of this anyway.